zerver/tests/test_events.py

```diff
@@ -1038,18 +1040,61 @@ def test_invitation_accept_invite_event(self) -> None:
             streams.append(get_stream(stream_name, self.user_profile.realm))
 
         do_invite_users(self.user_profile, ["foo@zulip.com"], streams, False)
-        prereg_users = PreregistrationUser.objects.get(email="foo@zulip.com")
+        prereg_user = PreregistrationUser.objects.get(email="foo@zulip.com")
 
         events = self.do_test(
             lambda: do_create_user('foo@zulip.com', 'password', self.user_profile.realm,
-                                   'full name', 'short name', prereg_user=prereg_users),
+                                   'full name', 'short name', prereg_user=prereg_user),
             state_change_expected=True,
             num_events=5,
         )
 
         error = schema_checker('events[4]', events[4])
         self.assert_on_error(error)
 
+        key = generate_key()
+        Confirmation.objects.create(content_object=prereg_user,
+                                    date_sent=timezone_now(),
+                                    confirmation_key=key,
+                                    type=Confirmation.INVITATION)
+
+        self.assertEqual(prereg_user.status, 0)
+        url = confirmation_url(key, self.user_profile.realm.host, Confirmation.INVITATION)
+        response = self.client_get(url)
+
+        prereg_user.refresh_from_db()
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(prereg_user.status, 1)
+
+    def test_invitation_accept_as_staff_with_support_invite_event(self) -> None:
+        self.user_profile = self.example_user('iago')
+        self.user_profile.is_staff = True
+        self.login(self.example_email('iago'))
+
+        streams = []
+        for stream_name in ["Denmark", "Scotland"]:
+            streams.append(get_stream(stream_name, self.user_profile.realm))
+
+        do_invite_users(self.user_profile, [self.nonreg_email('alice')], streams, False)
+        prereg_user = PreregistrationUser.objects.create(email=self.nonreg_email('alice'),
+                                                         referred_by=self.user_profile,
+                                                         realm=self.user_profile.realm)
+
+        key = generate_key()
+        Confirmation.objects.create(content_object=prereg_user,
+                                    date_sent=timezone_now(),
+                                    confirmation_key=key,
+                                    type=Confirmation.INVITATION)
+
+        self.assertEqual(prereg_user.status, 0)
+        url = confirmation_url(key, self.user_profile.realm.host, Confirmation.INVITATION)
+        url += "?support=True"
+        response = self.client_get(url)
+
+        prereg_user.refresh_from_db()
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(prereg_user.status, 0)
```

Comment:
Move this test as well to test_signup. Also don't stop the test here. Complete the whole sign up/invite process and at the end of sucessful regsutration verify that the status is still 0.